nio/client/base_client.py

```diff
@@ -620,6 +621,18 @@ def _handle_joined_members(self, response):
         if room.encrypted and self.olm is not None:
             self.olm.update_tracked_users(room)
 
+    def _handle_room_forget_response(self, response):
+        self.encrypted_rooms.discard(response.room_id)
+
+        if response.room_id in self.rooms:
+            room = self.rooms.pop(response.room_id)
+
+            if room.encrypted and self.store:
```

Comment:
If we only remove rooms here the room won't get removed if some other client calls the forget API endpoint but I'm unsure if there is a way to know that a room has been forgotten from a sync response. I guess no since the js-sdk seems to do the same thing. Do you know perhaps more about this?

Reply:
> If we only remove rooms here the room won't get removed if some other client calls the forget API endpoint but I'm unsure if there is a way to know that a room has been forgotten from a sync response. I guess no since the js-sdk seems to do the same thing. Do you know perhaps more about this?

The client-server API docs only say this:
"A user can also forget a room which they have left. Rooms which have been forgotten will never appear the response to the /sync API, until the user re-joins or is re-invited."

I tested forgetting a room with my client yesterday, and while the response was successful, Riot didn't notice the change. Today, it won't show me the "Historical" rooms category for some reason, so I don't really know.

Reply:
Right, I guess that's just the way it works. Clients need to forget rooms for themselves.